### PR TITLE
Implement moveResource -- Address #9

### PR DESCRIPTION
**Pull request by ruebot**
See: #9

Still need to get TravisCI hooked up proper.

You can see the build [here](https://travis-ci.org/ruebot/chullo/builds/89975877). It appears to be passing. But, same as with #19 & #20, I'm not confident I'm doing things right here.

_tags_ @Islandora-CLAW/7-x-2-x-committers

### CODE REVIEW

**Comment by DiegoPino** [comment]
@ruebot I think you are fine :smile, still have to look at the tests, but a new question arises(pinging @daniel-dgi also) , should we be also using a transactions aware `$uri` for move? So building `$uri` like in other methods and passing $transaction as argument also.
  `$uri = $this->prepareUri($uri, $transaction);` 

Not sure, since after a Move the base `$uri` would have changed, does transactions keep on with that?

File: src/IFedoraClient.php [hunk 1]
```diff
@@ -133,6 +133,16 @@ public function modifyResource($uri,
...
                                    $transaction = "");
+    /**
+     * Issues a MOVE request to Fedora.
+     *
+     * @param string    $uri            Resource URI
+     * @param array     $destination    Destination URI
+     *
+     * @return string
+     */
+    public function moveResource($uri,
```
Comment (daniel-dgi) [inline src/IFedoraClient.php]: Expose transactionality here as well, just like in #20

**Comment by daniel-dgi** [comment]
I'll have to run some tests manually with curl to see what happens within a transaction.  We may want to include the transaction in the destination as well.  I'll let you know what I find.

File: src/Chullo.php [hunk 2]
```diff
@@ -310,6 +310,37 @@ protected function prepareUri($uri, $transaction = "") {
...
+     *
+     * @return string
+     */
+    public function moveResource($uri,
+                                 $destination,
+                                 $transaction = "") {
+        // Ensure uri takes transaction into account.
+        $uri = $this->prepareUri($uri, $transaction);
+        // Create destinsation URI
+        $destination_uri = "Destination: " . $destination;
```
Comment (daniel-dgi) [inline src/Chullo.php]: You'll need to prepare this URI as well.

File: src/Chullo.php [hunk 3]
```diff
@@ -310,6 +310,37 @@ protected function prepareUri($uri, $transaction = "") {
...
+     *
+     * @return string
+     */
+    public function moveResource($uri,
+                                 $destination,
+                                 $transaction = "") {
+        // Ensure uri takes transaction into account.
+        $uri = $this->prepareUri($uri, $transaction);
+        // Create destinsation URI
+        $destination_uri = "Destination: " . $this->prepareUri($destination, $transaction);
```
Comment (daniel-dgi) [inline src/Chullo.php]: This results in 502.  No need to manually append 'Destination' as a string.  Just use guzzle's options array.  You need to nest the headers in there.  Looks something like this.
```php
        $destination_uri = $this->prepareUri($destination, $transaction);
        // Create destination array
        $options = [
          'headers' => [
            'Destination' => $destination_uri,
            'Overwrite'   => 'T'
          ],
        ];
        $response = $this->client->request(
          'MOVE',
          $uri,
          $options
        );
```

File: src/Chullo.php [hunk 4]
```diff
@@ -310,6 +310,39 @@ protected function prepareUri($uri, $transaction = "") {
...
+    public function moveResource($uri,
+                                 $destination,
+                                 $transaction = "") {
+        // Ensure uri takes transaction into account.
+        $uri = $this->prepareUri($uri, $transaction);
+        // Create destinsation URI
+        $destination_uri = $this->prepareUri($destination, $transaction);
+        // Create destination array
+        $options = [
+          'header' => [
```
Comment (daniel-dgi) [inline src/Chullo.php]: Needs to be 'headers'

**Comment by daniel-dgi** [comment]
Tested.  Good to go.

**Pull request merged**